plugin/src/main/java/org/opensearch/ml/action/connector/DeleteConnectorTransportAction.java

```diff
@@ -44,104 +52,175 @@
+        }
+    }
+
+    private void handleConnectorAccessValidationFailure(String connectorId, Exception e, ActionListener<DeleteResponse> actionListener) {
+        log.error("Failed to delete ML connector: {}", connectorId, e);
+        actionListener.onFailure(e);
+    }
+
+    private void checkForModelsUsingConnector(String connectorId, String tenantId, ActionListener<DeleteResponse> actionListener) {
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            ActionListener<DeleteResponse> restoringListener = ActionListener.runBefore(actionListener, context::restore);
+            SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+            sourceBuilder.query(QueryBuilders.matchQuery(MLModel.CONNECTOR_ID_FIELD, connectorId));
+            if (mlFeatureEnabledSetting.isMultiTenancyEnabled()) {
+                sourceBuilder.query(QueryBuilders.matchQuery(TENANT_ID_FIELD, tenantId));
+            }
+
+            SearchDataObjectRequest searchDataObjectRequest = SearchDataObjectRequest
+                .builder()
+                .indices(ML_MODEL_INDEX)
+                .tenantId(tenantId)
+                .searchSourceBuilder(sourceBuilder)
+                .build();
+            sdkClient
+                .searchDataObjectAsync(searchDataObjectRequest, client.threadPool().executor(GENERAL_THREAD_POOL))
+                .whenComplete((sr, st) -> {
+                    if (sr != null) {
+                        try {
+                            SearchResponse searchResponse = SearchResponse.fromXContent(sr.parser());
+                            SearchHit[] searchHits = searchResponse.getHits().getHits();
+                            if (searchHits.length == 0) {
+                                deleteConnector(connectorId, tenantId, restoringListener);
+                            } else {
+                                handleModelsUsingConnector(searchHits, connectorId, restoringListener);
                             }
-                            actionListener
+                        } catch (Exception e) {
+                            log.error("Failed to parse search response", e);
+                            restoringListener
                                 .onFailure(
-                                    new OpenSearchStatusException(
-                                        searchHits.length
-                                            + " models are still using this connector, please delete or update the models first: "
-                                            + Arrays.toString(modelIds.toArray(new String[0])),
-                                        RestStatus.CONFLICT
-                                    )
+                                    new OpenSearchStatusException("Failed to parse search response", RestStatus.INTERNAL_SERVER_ERROR)
```

Comment:
If the response has this exception message `Failed to parse search response`, user might be confused as to why search is being invoked in delete.